src/rest-server/src/models/v2/job/k8s.js

```diff
@@ -310,6 +292,33 @@ const generateTaskRole = (taskRole, labels, config) => {
       minSucceededTaskCount: -1,
     };
   }
+  // hived spec
+  if (hivedConfig.enabledHived) {
+    frameworkTaskRole.task.pod.spec.schedulerName = launcherConfig.scheduler;
+
+    delete frameworkTaskRole.task.pod.spec.containers[0].resources.limits['nvidia.com/gpu'];
+    frameworkTaskRole.task.pod.spec.containers[0]
+      .resources.limits['hivedscheduler.microsoft.com/pod-scheduling-enable'] = 1;
+
+    if (frameworkTaskRole.task.pod.metadata.annotations == null) {
+      frameworkTaskRole.task.pod.metadata.annotations = {};
+    }
```

Comment:
unnecessary

Reply:
yeah, it's a redundent check to avoid unnecessary logic dependency. Anyway, removal is also reasonable, will remove them